tiledb/sm/subarray/subarray.cc

```diff
@@ -1121,16 +1121,28 @@ Status Subarray::set_coalesce_ranges(bool coalesce_ranges) {
 }
 
 Status Subarray::to_byte_vec(std::vector<uint8_t>* byte_vec) const {
-  if (range_num() != 1)
+  if (range_num() != 1) {
     return logger_->status(Status_SubarrayError(
         "Cannot export to byte vector; The subarray must be unary"));
+  }
 
   byte_vec->clear();
 
-  for (const auto& subset : range_subset_) {
-    auto offset = byte_vec->size();
-    byte_vec->resize(offset + subset[0].size());
-    std::memcpy(&(*byte_vec)[offset], subset[0].data(), subset[0].size());
+  if (range_subset_.size() > 0 && range_subset_[0].num_ranges() > 0) {
+    for (const auto& subset : range_subset_) {
+      auto offset = byte_vec->size();
+      byte_vec->resize(offset + subset[0].size());
+      std::memcpy(&(*byte_vec)[offset], subset[0].data(), subset[0].size());
+    }
+  } else {
+    for (const auto& subset : label_range_subset_) {
+      auto offset = byte_vec->size();
+      byte_vec->resize(offset + subset->ranges_[0].size());
+      std::memcpy(
+          &(*byte_vec)[offset],
+          subset->ranges_[0].data(),
+          subset->ranges_[0].size());
+    }
```

Comment:
Since you removed this from the `serialiazation/query.cc` we can just fully remove it now.